wire-protoc-compatibility-tests/src/main/proto/squareup/proto3/java/interop/test_proto3_optional.proto

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+syntax = "proto3";
+
+package squareup.proto3.java.interop;
+
+import "google/protobuf/duration.proto";
+
+/*
+ * The name of this file must contain `test_proto3_optional`. From: the protobuf site,
+ * https://github.com/protocolbuffers/protobuf/blob/master/docs/implementing_proto3_presence.md
+ *
+ * > There are two options for getting around this error:
+ * >
+ * > 1. Pass --experimental_allow_proto3_optional to protoc.
```

Comment:
By the way it looks like the Protobuf Gradle plugin doesn't plan on adding it so our only option is to put `test_proto3_optional ` into the file name as you did.